core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceCreator.scala

```diff
@@ -0,0 +1,688 @@
+                                              span: Span = null): Future[Iterable[Any]] = {
+    traceWithParent("traversePolicies", span)(s1 =>
+      Future.traverse(policyNameToEmailMap) { x =>
+        val policyName = x._1
+        if (policyName == SamWorkspacePolicyNames.projectOwner && authDomains.isEmpty) {
+          // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
+          // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
+          // the limit of 2000
+          Future.successful(())
+        } else if (WorkspaceAccessLevels.withPolicyName(policyName.value).isDefined) {
+          // only sync policies that have corresponding WorkspaceAccessLevels to google because only those are
+          // granted bucket access (and thus need a google group)
+          traceWithParent(s"syncPolicy-${policyName}", s1)(_ => samDAO.syncPolicyToGoogle(SamResourceTypeNames.workspace, workspace.workspaceId, policyName))
+        } else {
+          Future.successful(())
+        }
+      }
+    )
+  }
+
+  private def setupGoogleProjectForWorkspace(workspace: Workspace,
+                                             policyNameToEmailMap: Map[SamResourcePolicyName, WorkbenchEmail],
+                                             billingProjectOwnerPolicyEmail: WorkbenchEmail,
+                                             authDomains: Set[ManagedGroupRef],
+                                             span: Span = null): Future[GoogleWorkspaceInfo] = {
+    //there's potential for another perf improvement here for workspaces with auth domains. if a workspace is in an auth domain, we'll already have
+    //the projectOwnerEmail, so we don't need to get it from sam. in a pinch, we could also store the project owner email in the rawls DB since it
+    //will never change, which would eliminate the call to sam entirely
+    val policyEmails = policyNameToEmailMap.map { case (policyName, policyEmail) =>
+      if (policyName == SamWorkspacePolicyNames.projectOwner && authDomains.isEmpty) {
+        // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
+        // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
+        // the limit of 2000
+        Option(WorkspaceAccessLevels.ProjectOwner -> billingProjectOwnerPolicyEmail)
+      } else {
+        WorkspaceAccessLevels.withPolicyName(policyName.value).map(_ -> policyEmail)
+      }
+    }.flatten.toMap
+
+    traceWithParent("gcsDAO.setupWorkspace", span)(_ => gcsDAO.setupWorkspace(userInfo, workspace.googleProjectId, policyEmails, workspace.bucketName, getLabels(authDomains.toList), span))
+  }
+
+  private def getLabels(authDomain: List[ManagedGroupRef]): Map[String, String] = authDomain match {
```

Comment:
get what kind of labels?

Reply:
Look like labels placed on the bucket.  Again, this was just code I copied over and didn't dig into unless I needed to.  Once again, if you want to clean this up now, just say the words otherwise I'm inclined to leave it.